### PR TITLE
fix: warp route deployment verification

### PR DESCRIPTION
**Pull request by paulbalaji**
fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3457

should fix the problem whilst also not breaking the format of existing artifact read/writes

drive-by fix to contractverifier - allows propagation of apikeys from custom chain metadata (e.g. chains.yaml)

### CODE REVIEW

File: typescript/cli/src/deploy/warp.ts [hunk 1]
```diff
@@ -255,7 +257,7 @@ async function executeDeploy(params: DeployParams) {
 
   const [contractsFilePath, tokenConfigPath] = prepNewArtifactsFiles(outPath, [
     { filename: 'warp-route-deployment', description: 'Contract addresses' },
-    { filename: 'warp-ui-token-config', description: 'Warp UI token config' },
+    { filename: 'warp-config', description: 'Warp config' },
```
Comment (yorhodes) [inline typescript/cli/src/deploy/warp.ts:260]: why rename this? this shape is different from the warp deploy config

File: typescript/sdk/src/deploy/verify/ContractVerifier.ts [hunk 2]
```diff
@@ -73,7 +73,11 @@ export class ContractVerifier {
     verificationLogger: Debugger,
     options?: FormOptions<typeof action>,
   ): Promise<any> {
-    const { apiUrl, family } = this.multiProvider.getExplorerApi(chain);
+    const {
+      apiUrl,
+      apiKey = this.apiKeys[chain],
```
Comment (yorhodes) [inline typescript/sdk/src/deploy/verify/ContractVerifier.ts:78]: this is slightly hard to read

File: typescript/sdk/src/token/contracts.ts
Comment (yorhodes) [inline typescript/sdk/src/token/contracts.ts:1]: honestly I would be happy to change the serialization format from
`[TokenType]: address` to `[ContractName]: address`, which is more consistent with the rest of the codebase

Comment (yorhodes) [reply typescript/sdk/src/token/contracts.ts:1]: (albeit this is a breaking change)

File: typescript/cli/src/deploy/warp.ts [hunk 3]
```diff
@@ -316,40 +318,47 @@ function getTokenName(token: TokenConfig) {
...
   }> = objMap(contracts, (chain, contract) => {
+    const tokenType = configMap[chain].type;
+    const contractName = isNft
+      ? hypERC721contracts[tokenType as keyof typeof hypERC721contracts]
+      : hypERC20contracts[tokenType as keyof typeof hypERC20contracts];
     return {
-      router: contract[configMap[chain].type as keyof TokenFactories].address,
-      tokenType: configMap[chain].type,
+      router: contract[contractName as keyof TokenFactories]['address'],
+      tokenType,
```
Comment (yorhodes) [inline typescript/cli/src/deploy/warp.ts:333]: honestly im not a fan of this different artifact shape
its another unnecessary layer of indirection

Comment (yorhodes) [reply typescript/cli/src/deploy/warp.ts:333]: [ContractSourceName]: address is more desirable
is this format recovered too?

**Review by yorhodes — APPROVED** [review]
looks good
please add a changeset

**Review by ltyu — APPROVED** [review]

File: typescript/cli/src/deploy/warp.ts (hunk 1 above)
Comment (paulbalaji) [reply typescript/cli/src/deploy/warp.ts:260]: yeah it's already `warp-config` on main, unsure why it's showing up as a change here

**Comment by paulbalaji** [comment]
created an alternative fix that doesn't change any of the warp artifact logic, simply creates a `deployContractWithName` that lets you override the contractName used at deploy time without changing the factory key 
https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3494

**Pull request closed without merging**